specs/phase0/beacon-chain.md

```diff
@@ -1724,13 +1727,13 @@ def process_deposit(state: BeaconState, deposit: Deposit) -> None:
     assert is_valid_merkle_branch(
         leaf=hash_tree_root(deposit.data),
         branch=deposit.proof,
-        depth=DEPOSIT_CONTRACT_TREE_DEPTH + 1,  # Add 1 for the List length mix-in
+        depth=uint64(DEPOSIT_CONTRACT_TREE_DEPTH + 1),  # Add 1 for the List length mix-in
         index=state.eth1_deposit_index,
         root=state.eth1_data.deposit_root,
     )
 
     # Deposits must be processed in order
-    state.eth1_deposit_index += 1
+    state.eth1_deposit_index = uint64(state.eth1_deposit_index + 1)
```

Comment:
Hmmm it's so strange. `exit_queue_epoch += Epoch(1)` is okay, but `state.eth1_deposit_index += uint64(1)` returns error:
> Incompatible types in assignment (expression has type "uint", variable has type "uint64")

Strange dark mypy hole. 🤦‍♀️

Reply:
The integer additions etc. are implemented on the uint base class in remerkleable, to not duplicate it everywhere. I will try to add typevar annotations that might help mypy understand it, then you can try this in the spec.

Reply:
Would this help? https://github.com/protolambda/remerkleable/compare/uint_type

Reply:
If that epoch thing works, then `state.eth1_deposit_index += DepositIndex(1)` should also work.

Reply:
Oh nvm, DepositIndex doesn't have its own type in the pyspec. Hmm, that error is weird.

Reply:
I still wonder about this thing, but suppose the other typing fixes outweigh this style issue.